### PR TITLE
feat(docker): add Dockerfile

### PR DESCRIPTION
**Pull request by mritd**
add Dockerfile

Signed-off-by: mritd <mritd1234@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    && apk add --virtual .build-deps \
+        build-base \
+        boost-dev \
+        curl \
+        cmake \
+        mariadb-dev \
+        tar \
+        git \
+    && curl -sSL ${LINUX_HEADERS_DOWNLOAD_URL} > /linux-headers-4.4.6-r2.apk \
+    && apk add --virtual .build-deps-kernel /linux-headers-4.4.6-r2.apk \
```
Comment (mritd) [inline Dockerfile:26]: Use a lower version of the kernel header for TFO to be compatible with most docker host systems.

refs: mritd/dockerfile#73

**Comment by GreaterFire** [comment]
Thank you for your contribution to the trojan project! We've already had a Dockerfile in [this repo](https://github.com/trojan-gfw/trojan-docker). If you want to make Docker related change, please file a pull request against that repository.

**Comment by GreaterFire** [comment]
Also I don't suggest lowering the kernel header version to support `TFO`. [The build flag `-DFORCE_TCP_FASTOPEN=ON`](https://trojan-gfw.github.io/trojan/build) is specifically for that.

**Comment by mritd** [comment]
Oh sorry, I didn't notice the dockerfile repo...

Suggestions for turning on TFO support by default:

Due to the docker operating mechanism (shared host kernel), if we open TFO compilation under alpine 3.9, it may cause many users to be unable to run on the actual system (eg: centos7, Ubuntu 16/18); because alpine 3.9 latest kernel header The version is 4.18, and most mainstream systems have not yet been upgraded to a high version.

So do we need to turn off TFO by default?

**Comment by GreaterFire** [comment]
@mritd Actually, we don't have to care about this problem. We can always turn TFO off in the config file, instead of at compile time. Also, if the host machine doesn't support TFO and we turn it on, trojan will ignore (well, kinda) the config anyways.

**Comment by mritd** [comment]
ok, i close this issue.

**Pull request closed without merging**